### DISCUSSION TITLE
About the cache patch to fetch

### DISCUSSION THREAD

**Discussion by stefan-garofalo** [vercel/next.js#65405] [Help] [answered]
### Summary

As stated in this React's [PR](https://github.com/facebook/react/pull/28896) fetch will no longer automatically cache, or rather, **dedupe** requests with same paramateres. Theo made a [video](https://www.youtube.com/watch?v=AKNH7mXciEM) already about it, which sums up the situation enough. 

Apart from the reasons behind the PR I'm concerned about a core feature in Nextjs being lost. Wrapping with `React.cache` every fetch is both unhandy, considered how we got used to this automatic behavior, and hard to recover. It's kind of a breaking change considering we all would have to path our codebases and wrap every fetch call when the new React's version comes out.

**My question is**: 
Can Nextjs itself patch fetch? 
Or must it be strongly bonded with React fetch implementation?

Again, since its pretty breaking I am a bit concerned about this, I would like not to go over all my repositories to recreate this lost behavior. What is Nextjs going to do about this?

**Comment by stefan-garofalo** [comment]
I found this PR https://github.com/vercel/next.js/pull/65058.
Do I understand correctly that Nextjs is now independently patching fetch to dedupe apart from React?

**Comment by icyJoseph** [comment, marked as answer]
Hi,

In the video you link, I don't think they really understood what was going on.

Yes React did patch fetch, however, Next.js made its own fetch patch:

- https://github.com/vercel/next.js/blob/canary/packages/next/src/server/lib/incremental-cache/index.ts#L305

As you can see, it even cross-referenced the React patch. This has been the case for a while now, the git blame shows commits going back as far back as 2 years.

Which means that even though React itself is dropping that kind of caching, Next.js does have it. It has been a so called 'user-land' patch.

**Reply by stefan-garofalo** [reply]
Great, so we are not loosing this behavior!

**Reply by neos04** [reply]
so why are my data fetches no longer being cached?

**Reply by icyJoseph** [reply]
Version? Type of request? In Next 15 they won't automatically cache anymore, you have to opt-in.
